### ISSUE TITLE
npm install still fails for v3.2.0 and Node.js 12

### ISSUE THREAD

**Issue by gabmontes** [OpenZeppelin/openzeppelin-contracts#2361]
Just cloning the repository and running `npm install` results in an error. This was previously reported in #1930 and supposedly fixed with #1931 but the problem persists.

**💻 Environment**

macOS 11 Big Sur
Node.js v12.18.1
openzeppelin-contracts@v3.2.0

**🔢 Code to reproduce bug**

```sh
git clone https://github.com/OpenZeppelin/openzeppelin-contracts.git -b v3.2.0
cd openzeppelin-contracts
npm i
```

Output:

```
10 warnings and 10 errors generated.
make: *** [Release/obj.target/sha3/src/addon.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (~/.nvm/versions/node/v12.18.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 20.1.0
gyp ERR! command "~/.nvm/versions/node/v12.18.1/bin/node" "~/.nvm/versions/node/v12.18.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd ~/openzeppelin-contracts/node_modules/ganache-core-coverage/node_modules/sha3
gyp ERR! node -v v12.18.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: scrypt@6.0.3 (node_modules/ganache-core-coverage/node_modules/scrypt):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: scrypt@6.0.3 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sha3@1.2.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sha3@1.2.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2020-09-15T00_55_05_716Z-debug.log
```

**Comment by abcoathup** [comment]
Hi @gabmontes!  

I’m sorry that you had this issue.  

We have been able to reproduce this issue by following these steps:
On WSL2 Ubuntu 18.04
```console
$ nvm install v12.18.1
$ git clone https://github.com/OpenZeppelin/openzeppelin-contracts.git -b v3.2.0
$ cd openzeppelin-contracts/
$ npm i
```
```console
sha3.target.mk:115: recipe for target 'Release/obj.target/sha3/src/addon.o' failed
make: *** [Release/obj.target/sha3/src/addon.o] Error 1
make: Leaving directory '/home/abcoathup/projects/forum/issue2361/openzeppelin-contracts/node_modules/ganache-core-coverage/node_modules/sha3/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/abcoathup/.nvm/versions/node/v12.18.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Linux 4.19.84-microsoft-standard
gyp ERR! command "/home/abcoathup/.nvm/versions/node/v12.18.1/bin/node" "/home/abcoathup/.nvm/versions/node/v12.18.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/abcoathup/projects/forum/issue2361/openzeppelin-contracts/node_modules/ganache-core-coverage/node_modules/sha3
gyp ERR! node -v v12.18.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.11 (node_modules/live-server/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.11: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.11 (node_modules/ganache-core/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.11: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules/ganache-core/node_modules/mocha/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: scrypt@6.0.3 (node_modules/ganache-core-coverage/node_modules/scrypt):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: scrypt@6.0.3 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sha3@1.2.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sha3@1.2.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/abcoathup/.npm/_logs/2020-09-16T05_09_17_970Z-debug.log
```

Thanks so much for reporting it! The project owner will review and triage this issue as soon as they can.  In the meantime, you can try the following workaround:

Use node 10 (rather than node 12).  I can `git clone` and `install` using node `v10.21.0`

If you just want to use OpenZeppelin Contracts then you can install in your project:
https://docs.openzeppelin.com/contracts/3.x/#overview

**Comment by frangio** [comment]
Thank you @gabmontes. This should be fixed now in `master`. I'm able to install everything using Node 12.8.4.

**Comment by lhumea6** [comment]
Broken for node v12.16.X, same error except this time it's the scrypt PBKDF dependency that fails to build

**Comment by 0xM4st3R** [comment]
Same error why you close it even the error exists @frangio

**Comment by frangio** [comment]
@0xM4st3R I closed this issue when it was fixed in Node 12.8.4.

**Comment by frangio** [comment]
This is fixed. I'm able to run `npm install` on Node 12.19.0.

Note that there are `node-gyp` errors but they are not fatal, you should see a message that says "Keccak bindings compilation fail. Pure JS implementation will be used."

If you still see errors, please open a new issue specifying the exact Node version you're using.
